### DISCUSSION TITLE
GPIO controlled shortkey for favorite sounds

### DISCUSSION THREAD

**Discussion by Gitarrenzupfer** [probonopd/MiniDexed#722] [Ideas]
Hello,
I would like to build a Minidexed box that meets my personal requirements. I usually only use a few sounds and would like to have the option of using "favorite buttons" (4-8 sounds). Is this already possible via the minidexed.ini?

The only solution I can think of is to send the appropriate program change commands to the MIDI input via an Arduino Nano (or something similar).

Perhaps someone has already asked themselves the same question and has an idea...

Greetings

**Comment by miotislucifugis** [comment]
aside from pgm change... you can reorder your favorite performances to default and/ performance#1-4 of performance bank1.  then use the performance up/down gpio buttons to navigate them.   A little awkward because you have to eject the sd card and manually 
renumber/reoder the performances, every time your "favorites" change.      Sorta  what I was trying to start a discussion about here: https://github.com/probonopd/MiniDexed/discussions/715

**Comment by diyelectromusic** [comment]
I keep meaning to do a "DX voice button" style controller for MiiniDexed - I have kind of a Pico and arduino version - but yes, these are using a microcontroller to send PC messages - https://diyelectromusic.com/2022/05/20/pi-pico-midi-button-voice-select/

I doubt there are really enough GPIO to make it worth doing directly from a RPi without some kind of IO expander, which we don't really support atm (I had some prototype code here: https://github.com/probonopd/MiniDexed/discussions/509 but it didn't seem particularly wanted by anyone :))

Kevin

**Reply by probonopd** [reply]
Agree, I think this kind of functionality is best handled on something like a Pico. If I had to do it, I'd probably use MicroPython on the Pico for easy customizability.
